src/main/java/net/minecraftforge/fluids/FluidUtil.java

```diff
@@ -370,6 +372,34 @@ public static IFluidHandlerItem getFluidHandler(@Nonnull ItemStack itemStack)
     }
 
     /**
+     * Gets the fluid handler for a given fluid's block.
+     * The handler's fill method will place a fluid block with the appropriate amount when not simulated.
+     *
+     * @param world the world the block has to be placed in
+     * @param pos   the position the block has to be placed at
+     * @param fluid the fluid to place
+     * @return the fluid block's fluider handler
+     */
+    private static IFluidHandler getPlacementFluidHandler(final World world, final BlockPos pos, Fluid fluid)
```

Comment:
The problem with `getFluidHandler` (both the ItemStack and block versions) is that they are for blocks/containers that are already filled with some fluid. The purpose of this method, however, is to get a handler *for* a position *based* on the fluid that's supposed to be in that position.

Together with the comment below, I suppose it's better to move this method closer to `tryPlaceFluid` to signify the connection. Another option would be completely inlining the handler  construction, given that `tryPlaceFluid` is the only user of the method and it's very specific to it as well.